core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1193,7 +1189,50 @@ public void updateTask(TaskResult taskResult) {
                     task.getTaskDefName(), lastDuration, false, task.getStatus());
         }
 
-        _decide(workflowId);
+        // evaluate workflow only if the task is not within a forked branch
+        if (!isLazyEvaluateWorkflow(workflowInstance.getWorkflowDefinition(), task)) {
+            _decide(workflowId);
```

Comment:
What do you think of push the workflow into decide queue with high priority if we are not doing decide here so to speed up it to be evaluated by sweeper next

Reply:
good idea, the perceived delay would be lower by evaluating with a higher priority.